SCIInstitute/ShapeWorks
# Multidomain usecases and other changes

This PR addresses : 
#1450 Clean up code in sw.data and include find reference python function for multiple domains
#1446 Generate ellipsoid joints with random centers for each joint and minor changes to cohort generation

As per discussion with @sheryjoe  today: This PR add's domain specific alignment for the multiple domain segmentation use case. 
For the multiple domain data we currently have, we can't use global alignment as we want to preserve rotation w.r.t one ellipsoid as the mode of variation. Added inline comments for global alignment. Global alignment will be added for left atrium with appendage dataset in the next release)

To test:
1. ellipsoid joint generation - ellispoids joints should have random centers and, meshes should be saved in only 'vtk' format
```
import ShapeCohortGen
ellipsoid_generator = ShapeCohortGen.EllipsoidJointsCohortGenerator(out_dir="ellipsoid_joint_new")
ellipsoid_generator.generate(num_samples=3,mode_size=False,mode_rotation=True,randomize_center=True)
ellipsoid_generator.generate_segmentations(randomize_size=False, allow_on_boundary=False)
```
2. multiple domain sub-sampling and alignment - check if sub-sampling works and the model produced at the end is reasonable 
```
python RunUseCases.py ellipsoid_multiple_domain --use_subsample --num_subsample 3
```

3. All other use cases should work correctly with sub-sampling:
```
python RunUseCases.py ellipsoid --use_subsample --num_subsample 3
python RunUseCases.py ellipsoid_cut --use_subsample --num_subsample 3
python RunUseCases.py ellipsoid_mesh --use_subsample --num_subsample 3
python RunUseCases.py ellipsoid_multiple_domain_mesh --use_subsample --num_subsample 3
python RunUseCases.py femur --use_subsample --num_subsample 3
python RunUseCases.py femur_cut --use_subsample --num_subsample 3
python RunUseCases.py femur_mesh --use_subsample --num_subsample 3
python RunUseCases.py left_atrium --use_subsample --num_subsample 3
python RunUseCases.py lumps --use_subsample --num_subsample 3
```
